packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts

```diff
@@ -337,12 +336,24 @@ export class AuthTransformer extends TransformerAuthBase implements TransformerA
       // get fields specified in the schema
       // if there is a role that does not have read access on the field then we create a field resolver
       // or there is a relational directive on the field then we should protect that as well
-      const readRoles = acm.getRolesPerOperation('read');
+      const readRoles = [...new Set([
+        ...acm.getRolesPerOperation('get'),
+        ...acm.getRolesPerOperation('list'),
+        ...acm.getRolesPerOperation('sync'),
+        ...acm.getRolesPerOperation('search'),
+        ...acm.getRolesPerOperation('listen'),
+      ])];
       const modelFields = def.fields?.filter(f => acm.hasResource(f.name.value)) ?? [];
       const errorFields = new Array<string>();
       modelFields.forEach(field => {
         const fieldReadRoles = getReadRolesForField(acm, readRoles, field.name.value);
-        const allowedRoles = fieldReadRoles.filter(r => acm.isAllowed(r, field.name.value, 'read'));
+        const allowedRoles = fieldReadRoles.filter(
+          r => acm.isAllowed(r, field.name.value, 'get')
+                || acm.isAllowed(r, field.name.value, 'list')
+                || acm.isAllowed(r, field.name.value, 'search')
+                || acm.isAllowed(r, field.name.value, 'listen')
+                || acm.isAllowed(r, field.name.value, 'sync'),
+        );
```

Comment:
should we move this logic into its own function? In that way, this can be reused for `roleDefinitions` assignment as well.